core/src/main/scala/kafka/log/Log.scala

```diff
@@ -746,6 +753,7 @@ class Log(@volatile var dir: File,
     maybeHandleIOException(s"Exception while increasing log start offset for $topicPartition to $offset in dir ${dir.getParent}") {
       lock synchronized {
         if (offset > logStartOffset) {
+          info(s"Incrementing log start offset of partition $topicPartition to $offset in dir ${dir.getParent}")
```

Comment:
Just confirming since this is at info level: it will be logged when segments are deleted or when records at the head of the log are purged, right? If so, info seems like a reasonable log level for now.